### PR TITLE
pkp/pkp-lib#730 automatic DOI registration to DataCite

### PR DESCRIPTION
**Pull request by bozana**
S. https://github.com/pkp/pkp-lib/issues/730

### CODE REVIEW

File: plugins/importexport/datacite/DataciteInfoSender.inc.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * @file plugins/importexport/datacite/DataciteInfoSender.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class DataciteInfoSender
+ * @ingroup plugins_importexport_datacite
+ *
+ * @brief Scheduled task to DOIs to the DataCite server.
```
Comment (beghelli) [inline plugins/importexport/datacite/DataciteInfoSender.inc.php]: "to send DOIs" maybe?

File: plugins/importexport/datacite/DataciteInfoSender.inc.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	 */
+	function _getJournals() {
+		$plugin =& $this->_plugin;
+		$journalDao =& DAORegistry::getDAO('JournalDAO'); /* @var $journalDao JournalDAO */
+		$journalFactory =& $journalDao->getJournals(true);
+
+		$journals = array();
+		while($journal =& $journalFactory->next()) {
+			$journalId = $journal->getId();
+			if (!$plugin->getSetting($journalId, 'enabled') && !$plugin->getSetting($journalId, 'automaticRegistration')) continue;
```
Comment (beghelli) [inline plugins/importexport/datacite/DataciteInfoSender.inc.php]: I think that OR makes more sense...

File: plugins/importexport/datacite/scheduledTasks.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  * This file describes the schedule task that will send deposits to DataCite in the background.
+  *
+  -->
+
+<!DOCTYPE scheduled_tasks SYSTEM "../../../lib/pkp/dtd/scheduledTasks.dtd">
+
+<scheduled_tasks>
+	<task class="plugins.importexport.datacite.DataciteInfoSender">
+		<descr>Send DOI deposit to DataCite server.</descr>
+		<frequency minute="0"/>
```
Comment (beghelli) [inline plugins/importexport/datacite/scheduledTasks.xml]: If this is used by the acron plugin with frequency 0, then every website request will trigger the scheduled task. I think that 24 hours might be a better value.

File: plugins/importexport/datacite/DataciteInfoSender.inc.php (hunk 2 above)
Comment (bozana) [inline plugins/importexport/datacite/DataciteInfoSender.inc.php]: yes, you are right, thanks!!!

File: plugins/importexport/datacite/scheduledTasks.xml (hunk 3 above)
Comment (bozana) [inline plugins/importexport/datacite/scheduledTasks.xml]: OK :-) Should this be then also changed for the CrossRef plugin?

**Comment by bozana** [comment]
I made the changes according to the comments... THANKS!!!

File: plugins/importexport/datacite/scheduledTasks.xml (hunk 3 above)
Comment (beghelli) [inline plugins/importexport/datacite/scheduledTasks.xml]: Ok, I checked the AcronPlugin code and it doesn't allow 0 values. It uses
24 hours instead. So you can leave it that way. :)

2015-09-09 11:02 GMT-03:00 bozana <notifications@github.com>:

**Comment by bozana** [comment]
THANKS @beghelli !!!
I hope this is the last change :-)

**Comment by bozana** [comment]
You mean for this pull request? -- Yes :-)

**Pull request merged**